pkg/tbtc/registry.go

```diff
@@ -156,6 +156,48 @@ func (wr *walletRegistry) getWalletByPublicKeyHash(
 	return wallet{}, false
 }
 
+// archiveWallet archives the wallet with the given public key hash. The wallet
+// data is removed from the wallet cache and the entire wallet storage directory
+// is moved to the archive directory.
+func (wr *walletRegistry) archiveWallet(
+	walletPublicKeyHash [20]byte,
+) error {
+	wr.mutex.Lock()
+	defer wr.mutex.Unlock()
+
+	var walletPublicKey *ecdsa.PublicKey
+
+	for _, value := range wr.walletCache {
+		if value.walletPublicKeyHash == walletPublicKeyHash {
+			// All signers belong to one wallet. Take the wallet public key from
+			//  the first signer.
+			walletPublicKey = value.signers[0].wallet.publicKey
+		}
+	}
```

Comment:
We can use `wr.getWalletByPublicKeyHash` and extract the public key from the returned `wallet` instance. Note that it also returns a boolean flag which can be used to check whether the node controls that wallet.

Reply:
I initially wanted to do that, but the problem is that both the new function (`wr.archiveWallet`) and `wr.getWalletByPublicKeyHash` lock the same mutex.
Therefore calling `wr.getWalletByPublicKeyHash` inside `wr.archiveWallet` will cause a deadlock.

The solution could be writing a new function `wr.getWalletByPublicKeyHashImpl` and use it in both `wr.getWalletByPublicKeyHash` and `wr.archiveWallet`. Or we could simply leave it as is.
WDYT @lukasz-zimnoch

Reply:
Yeah, you are right! Let's leave as is.